Development practices/Turbo frames.md

```diff
@@ -0,0 +1,195 @@
+```ruby
+  def edit
+    @author = Author.find(params[:id])
+  end
+
+  def update
+    @author = Author.find(params[:id])
+
+    if @author.update(author_params)
+      redirect_to authors_path
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+```
+
+NOTE: If you are using slim, make sure to name the views like `'index.html.slim'` instead of just `'index.slim'` because otherwise the form submissions won't work correctly!
+
+## Targeting navigation into or out of a frame
+Sometimes we may want the most links to operate within the frame context, but not all of them.
+
+### data-turbo-frame
+Let's say that we also want to be able to add a new author on the index page, without redirecting to a completely new page.
+This time we don't want to replace anything that's visible on the index page.
+We actually want to "append" the form for creating a new author to the page.
+
+If we used the same approach as for the inline editing, we would replace the link for creating a new user with the form. Maybe we don't want to do this because we don't want the form to appear at the same position where the button is, but somewhere else on the page. Luckily it is also done very easily with turbo frames.
+
+Index page:
+```ruby
+  h2 = 'Authors'
+  = link_to 'Add author', new_author_path, data: { 'turbo-frame': 'new_author' }
+
+  = turbo_frame_tag 'new_author'
+  - @authors.each do |author|
+    = turbo_frame_tag "edit_author_#{author.id}"
+      .row
+        .col
+          = author.first_name
+        .col
+          = author.last_name
+        .col
+          = link_to 'Edit', edit_author_path(author.id)
+```
+
+The first change to our index page is the link to the new_author_path. This link is different to the one for the edit_author_path, because `data: { 'turbo-frame': 'new_author' }` was added to it.
+
+The `data-turbo-frame` attribute can be added on non-frame elements to control from which frame was the link clicked or the form submitted.
+
+In our example, we want the link to act as if it was clicked inside of a turbo-frame with the `'new_author'` id.
+
+And that turbo-frame is the second change for the index page. It is just an empty frame which will be replaced with the response from the #new action, which may look like this:
+```ruby
+  = turbo_frame_tag 'new_author'
+    = simple_form_for @author do |f|
+      .row
+        .col
+          = f.input :first_name
+        .col
+          = f.input :last_name
+        .col
+          = f.button :submit
+```
+
+So we click on the link to add a new author, it acts as if it was clicked inside of a `'new_author'` frame because of the `data-turbo-frame` attribute. The response provided by the #new action will have its turbo-frame segment extracted and the content will replace the frame from on the index page.
+
+### target
+By default, navigation within a frame will target just that frame. But it can also drive another named frame by setting the `target` to the ID of that frame.
```

Comment:
please elaborate a bit more for this section cause I find it hard to understand/visualize what's happening.